client/state/site-settings/exporter/actions.js

```diff
@@ -83,28 +96,32 @@ export function advancedSettingsFail( siteId, error ) {
  * @param  {Number}   siteId  The ID of the site to export
  * @return {Function}         Action thunk
  */
-export function startExport( siteId ) {
-	return ( dispatch ) => {
+export function startExport( siteId, exportAll = true ) {
+	return ( dispatch, getState ) => {
 		if ( ! siteId ) {
 			return;
 		}
 
 		dispatch( {
 			type: EXPORT_START_REQUEST,
-			siteId: siteId
+			siteId,
+			exportAll,
 		} );
 
+		const state = getState();
+		const advancedSettings = exportAll ? null : prepareExportRequest( state, siteId );
+
 		const success =
 			() => dispatch( exportStarted( siteId ) );
 
 		const failure =
 			error => dispatch( exportFailed( siteId, error ) );
 
 		return wpcom.undocumented()
-			.startExport( siteId )
+			.startExport( siteId, advancedSettings )
 			.then( success )
```

Comment:
Changing the names is a good thing, not only because it fits into our code styling guides, but also because it maps more naturally to the JavaScript idioms. When I see `post_type` and other related naming in Calypso, I can't help but think we are making mistakes by tying our JavaScript code so tightly to the server. In other words, we wouldn't have ever named these things this way if it weren't for the fact that we're looking at the code with our PHP API glasses on.

We can extend this too, because there are more subtle oddities in the data. Why are we storing a date as a string at all? We could transform the string from the API into a `Date()` object and use it directly throughout the rest of the code. Of course, we would have to serialize it to store it in the Redux state tree, but that's not a problem because it would be normalized to JavaScript serialization and free from change.

```js
const postTypeFromApi = ( {
	post_type: type,
	post_start_date: startDate,
	post_end_date: endDate
} ) => ( {
	type,
	startDate: Date.parse( startDate ),
	endDate: Date.parse( endDate )
} );
```